### PR TITLE
[DOCS] Moves model snapshot resource definitions into APIs

### PR DESCRIPTION
**Pull request by lcawl**
Related to #49700 and https://github.com/elastic/stack-docs/pull/755

This PR removes the model snapshot definitions page (https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-snapshot-resource.html) and puts that information directly into the appropriate APIs.

It also adds some missing definitions (`model_bytes_exceeded` and `model_bytes_memory_limit`).

Preview:
* http://elasticsearch_50157.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-delete-snapshot.html
* http://elasticsearch_50157.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-get-snapshot.html
* http://elasticsearch_50157.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-revert-snapshot.html
* http://elasticsearch_50157.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-update-snapshot.html

### CODE REVIEW

File: docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc [hunk 1]
```diff
@@ -58,52 +63,136 @@ include::{docdir}/ml/ml-shared.asciidoc[tag=job-id-anomaly-detection]
...
+* `soft_limit`: The internal models require more than 60% of the configured
+memory limit and more aggressive pruning will be performed in order to try to
+reclaim space.
+--
+
+`model_size_stats`.`model_bytes`:::
+(long) An approximation of the memory resources required for this analysis.
+
+`model_size_stats`.`model_bytes_exceeded`:::
+TBD
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc]:
```suggestion
(long) The number of bytes over the high limit for memory usage at the last allocation failure.
```

File: docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc [hunk 2]
```diff
@@ -58,52 +63,136 @@ include::{docdir}/ml/ml-shared.asciidoc[tag=job-id-anomaly-detection]
...
+
+`model_size_stats`.`model_bytes_memory_limit`:::
+TBD
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc]: Per https://github.com/elastic/elasticsearch/pull/50057:
```suggestion
(long) The upper limit for memory usage, checked on increasing values.
```

File: docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/ml/anomaly-detection/apis/get-snapshot.asciidoc]: @edsavage I assume these definitions you provided in https://github.com/elastic/elasticsearch/pull/50057 work here too?

File: docs/reference/ml/anomaly-detection/apis/revert-snapshot.asciidoc [hunk 3]
```diff
@@ -24,7 +24,7 @@ Reverts to a specific snapshot.
 [[ml-revert-snapshot-desc]]
 ==== {api-description-title}
 
-The {ml} feature in {xpack} reacts quickly to anomalous input, learning new
+The {ml-features} reacts quickly to anomalous input, learning new
```
Comment (edsavage) [inline docs/reference/ml/anomaly-detection/apis/revert-snapshot.asciidoc]:
```suggestion
The {ml-features} react quickly to anomalous input, learning new
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 4]
```diff
@@ -895,8 +895,7 @@ end::model-plot-config[]
 
 tag::model-snapshot-id[]
 A numerical character string that uniquely identifies the model snapshot. For 
-example, `1491007364`. For more information about model snapshots, see
-<<ml-snapshot-resource>>.
+example, `1491007364`.
```
Comment (edsavage) [inline docs/reference/ml/ml-shared.asciidoc]:
```suggestion
example, `1575402236000 `.
```
(Or similar, to match the updated examples earlier)

**Review by edsavage — APPROVED** [review]
LGTM

**Pull request merged**